applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+/*
+ * Copyright(c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#ifndef _AMOD_API_H_
+#define _AMOD_API_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include "data_fifo.h"
+#include "aobj_api.h"
+
+/**
+ * @brief Modules private handle.
+ */
+struct handle;
+
+/**
+ * @brief Private context for the module.
+ */
+struct amod_context;
+
+/**
+ * @brief Modules private configuration.
+ */
+struct amod_configuration;
+
+/**
+ * @brief Module type relative to the module.
+ */
+enum amod_type {
+	/* The module type is undefined */
+	AMOD_TYPE_UNDEFINED = 0,
+
+	/* This is an input only module */
+	AMOD_TYPE_INPUT,
+
+	/* This is an output only module */
+	AMOD_TYPE_OUTPUT,
+
+	/* This is an input/output processing module */
+	AMOD_TYPE_IN_OUT
+};
+
+/**
+ * @brief Module state.
+ */
+enum amod_state {
+	/* Module state undefined */
+	AMOD_STATE_UNDEFINED = 0,
+
+	/* Module is in the configured state */
+	AMOD_STATE_CONFIGURED,
+
+	/* Module is in the running state */
+	AMOD_STATE_RUNNING,
+
+	/* Module is in the stopped state */
+	AMOD_STATE_STOPPED
+};
+
+/**
+ * @brief Private pointer to a modules functions.
+ */
+struct amod_functions {
+	/**
+	 * @brief  Function for opening a module.
+	 *
+	 *
+	 * @param handle  A handle to this module instance
+	 *
+	 * @return 0 if successful, error value
```

Comment:
```suggestion
	 * @return 0 if successful, error value
```
```suggestion
	 * @return 0 if successful, error otherwise
```

This is relevant for several places in this file